File: was/auth/src/main/java/org/example/domain/repository/MemberReasonRepository.java
package org.example.domain.repository;

import org.example.domain.entity.MemberReason;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.UUID;

public interface MemberReasonRepository extends JpaRepository<MemberReason, UUID> {

    List<MemberReason> findAllByMemberId(UUID memberId);
}

File: was/member/src/main/java/org/example/domain/repository/MemberReasonRepository.kt
package org.example.domain.repository

import org.example.domain.entity.MemberReason
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface MemberReasonRepository: JpaRepository<MemberReason, UUID> {

    fun findAllByMemberId(memberId: UUID): List<MemberReason>
}